### PR TITLE
✨ Redo config system with TOML

### LINKED ISSUE #60: [Feature Request]: Rename/simplify "data" folder
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Potential Enhancement

Upon installing FO, the installer creates a "data" folder. That name should be more descriptive

### How it could be implemented

Rename it to "Vanilla Installer Data" or similar. 
Does it even need to be a folder that contains two files? How about just _one_ .json or .cfg file?

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by osfanbuff63**
Fixes #60

Somewhat tested, but more testing would be appreciated.

**Current to-do:**
- [x] Fix theme in GUI not being used, instead locking to the system one
- [x] Fix fonts not changing
- [x] Fix path dialog not updating until the dialog is executed

### CODE REVIEW

**Comment by tuxinal** [comment]
Looks good to me, except the fact that gui crashes. I fixed the crash and tried to upload the .patch file here but github doesn't let me - _despite listing .patch files as a valid file to upload_ - 
Besides that it works as expected

File: vanilla_installer/theme.py [hunk 1]
```diff
@@ -80,16 +68,15 @@ def load() -> dict:
...
     """
-    return dark_theme if is_dark() else light_theme
+    return dark_theme if is_dark("dark") else light_theme
 
 
 def toggle() -> None:
     """Switches between dark and light theme."""
-    is_dark(to_dark=not is_dark())
-
+    is_dark(is_dark() == "dark")
```
Comment (tuxinal) [inline vanilla_installer/theme.py]:
```suggestion
    is_dark(is_dark() != "dark")
```

File: vanilla_installer/theme.py [hunk 2]
```diff
@@ -80,16 +68,15 @@ def load() -> dict:
     Returns:
         dict: The colors palette.
     """
-    return dark_theme if is_dark() else light_theme
+    return dark_theme if is_dark("dark") else light_theme
```
Comment (tuxinal) [inline vanilla_installer/theme.py]:
```suggestion
    return dark_theme if is_dark() == "dark" else light_theme
```

File: vanilla_installer/gui.py [hunk 3]
```diff
@@ -27,7 +27,8 @@
 )
```
Comment (tuxinal) [inline vanilla_installer/gui.py]: this import got reverted by the main branch merge in the previous commit
```suggestion
import platform
```

File: vanilla_installer/gui.py [hunk 4]
```diff
@@ -27,7 +27,8 @@
...
 
 # LOCAL
-from vanilla_installer import main, theme
+from vanilla_installer import config, main, theme
+from vanilla_installer.log import logger
 
 # ARGUMENTS
 FONT_FILE = pathlib.Path("data/font.txt").resolve()
```
Comment (tuxinal) [inline vanilla_installer/gui.py]: not necessary anymore
```suggestion
```

**Review by tuxinal — CHANGES_REQUESTED** [review]
Looks good to me besides these issues
the theme is stuck on dark mode and the platform import in gui.py got reverted in the last merge commit

File: vanilla_installer/theme.py (hunk 1 above)
Comment (osfanbuff63) [reply vanilla_installer/theme.py]: :ioa: lol

**Pull request merged**